src/revocation.rs

```diff
@@ -7,10 +7,151 @@ use crate::error;
+        .current_dir(dir)
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+    {
+        Ok(child) => child,
+        Err(e) => {
+            let msg = format!(
+                "ERROR: failed to run revocation action: {}, received: {}",
+                script, e
+            );
+            error!("{}", msg);
+            return Err(Error::Other(msg));
+        }
+    };
+
+    let msg = format!(
+        "ERROR: failed writing to stdin on revocation action: {:?}",
+        script
+    );
+    if let Err(e) = child
+        .stdin
+        .as_mut()
+        .expect(&msg)
+        .write_all(raw_json.get().as_bytes())
+    {
+        error!("{}", msg);
+        return Err(Error::Other(msg));
+    }
+
+    let output = match child.wait_with_output() {
+        Ok(output) => output,
+        Err(e) => {
+            let msg = format!("ERROR: failed to wait on child process while running revocation action: {:?}", script);
+            error!("{}", msg);
+            return Err(Error::Other(msg));
+        }
+    };
+
+    if !output.status.success() {
+        let code = match output.status.code() {
+            Some(code) => code.to_string(),
+            None => {
+                "no code; process likely terminated by signal".to_string()
+            }
+        };
+        let stdout = String::from_utf8(output.stdout)?;
+        let stderr = String::from_utf8(output.stderr)?;
+
+        let mut msg = format!(
+            "ERROR: revocation action {} returned with {}\n",
+            script, code
+        );
+
+        if !stdout.is_empty() {
+            msg = format!(
+                "{}{}",
+                msg,
+                format!(
+                    "ERROR: revocation action {} stdout: {:?}",
+                    script, stdout
+                )
+            );
+        }
+
+        if !stderr.is_empty() {
+            msg = format!(
+                "{}{}",
+                msg,
+                format!(
+                    "ERROR: revocation action {} stderr: {:?}",
+                    script, stderr
+                )
+            );
+        }
+
+        error!("{}", msg);
+        return Err(Error::Other(msg));
+    }
+
+    info!(
+        "{}",
+        format!("INFO: revocation action {:?} successful", script)
+    );
+    Ok(output)
+}
+
+/// Runs revocation actions received from tenant post-attestation
+pub(crate) fn run_revocation_actions(
+    json: Value,
+) -> Result<Vec<Result<Output>>> {
+    #[cfg(not(test))]
+    pretty_env_logger::init();
+
+    #[cfg(test)]
+    let mount = concat!(env!("CARGO_MANIFEST_DIR"), "/tests");
+
+    #[cfg(not(test))]
+    let mount = secure_mount::mount()?;
+    let unzipped = format!("{}/unzipped", mount);
+    let action_file = format!("{}/unzipped/action_list", mount);
+
+    let mut outputs = Vec::new();
+
+    if Path::new(&action_file).exists() {
```

Comment:
future: It may make sense to load this once rather than when we run revocations. However, I believe this is fine and may be what we want long term depending on the expectations of the users.